airflow/timetables/trigger.py

```diff
@@ -89,7 +89,7 @@ def next_dagrun_info(
             else:
                 next_start_time = self._align_to_next(restriction.earliest)
         else:
-            start_time_candidates = [self._align_to_next(DateTime.utcnow())]
+            start_time_candidates = [self._align_to_next(DateTime.utcnow().replace(second=0, microsecond=0))]
```

Comment:
This makes sense :thinking: 

Could you add a test for this change? (a test which fails without resetting seconds and microsecond and passes after applying your change)

Reply:
Nice one, yeah. That would definitely explain occasionally lost dag runs

Reply:
And yeah - having a test for that would be fantastic - if you need help @BohdanSemonov - we can suggest an approach here.

Reply:
Wouldn’t think still have problems if the scheduler is a tad too slow and fails to schedule the DAG in a minute? I still prefer the `_align_to_prev` fix over this.

Reply:
@uranusjr - could you please elaborate? Is there a fix in progress you are referring to? Or some new code to be written?  It wasn't too clear. I think it would be worth to consider fixing this problem for 2.7.0

Reply:
The linked issue already contains discussion before this fix was proposed.

Reply:
Yeah. There are 3 linked issues and each of them has tens of comments. Maybe that is obvious to you, but would be great to point to it. A link to specific comment would be great, otherwise it forces somoene looking here to read the entire conversation.

Reply:
(Just try to see the other side of people who are participating in multiple discussions and might not remember all of it).

Reply:
I believe he means the change shown in these 2 comments. Apologies for interjecting, but I am excited for this fix so wanted to help keep things moving.

https://github.com/apache/airflow/issues/27399#issuecomment-1638152940
https://github.com/apache/airflow/issues/27399#issuecomment-1638156248

Reply:
I think it relies on @uranusjr -> I am unable to assess if it's good enough or not at this stage and how to move forward.